### ISSUE TITLE
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks.

### ISSUE THREAD

**Issue by ghost** [react-static/react-static#1211]
Hey guys,

Lately I've been encountering a weird log during build :

```
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks.
```

It shows pretty much after every template exported :

```
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks. Please check usage of Babel plugin.
Time to render to string and extract { '$$typeof': Symbol(react.element),
  type: [Function: FinalComp],
  key: null,
  ref: null,
  props: {},
  _owner: null }
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks. Please check usage of Babel plugin.
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks. Please check usage of Babel plugin.
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks. Please check usage of Babel plugin.
Time to render to string and extract { '$$typeof': Symbol(react.element),
  type: [Function: FinalComp],
  key: null,
  ref: null,
  props: {},
  _owner: null }
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks. Please check usage of Babel plugin.
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks. Please check usage of Babel plugin.
Time to render to string and extract { '$$typeof': Symbol(react.element),
  type: [Function: FinalComp],
  key: null,
  ref: null,
  props: {},
  _owner: null }
[FLUSH CHUNKS]: Unable to find src/pages/PageManager in Webpack chunks. Please check usage of Babel plugin.
```

It doesn't seem to affect the build, and the site works fine (even without JS on so html is being rendered correctly).

Any idea on what's causing this ?

**Comment by nddery** [comment]
Unsure how I missed this issue but I just posted in the Spectrum chat about what seem quite similar: https://spectrum.chat/react-static/general/flush-chunks-warnings-errors-when-building-outside-of-react-static-root~311c52c1-bd17-45a1-b4d1-376337072681

Do you also modify the `config.paths.dist` config ?

**Comment by simonunlikely** [comment]
Hey,

I ran into your post I think, seems like the same problem but I did not modify the `config.paths.dist`.

My `static.config.js` is pretty standard, I'm just using `Document` and setting a bunch of routes.

I have absolutely no clue as to what this is all about, especially as it doesn't seem to break anything in the build. Is that the case for you too?

**Comment by nddery** [comment]
It also doesn't seem to break anything for us, but the warning goes away as soon as we remove the `config.path.dist` from the config.

**Comment by simonunlikely** [comment]
Hey,

I managed to pin down where this message is coming from, so I thought I'd let you guys know in case anyone is having the same issue :

In my app I'm using a `PageManager` component that is the template for every route in my app. This component is reading the route data, and based on the type, it chooses the corresponding page template to render. It works just fine.

BUT. I'm also importing this component in the top level `App` component and use it as a dynamic route. The purpose of this is to have dynamic previews of data from the CMS. When I hit the `/preview` url, it renders the `PageManager` dynamically and renders the template based on the preview data just as it does with normal routes.

The error message pops up whenever you import in `App` a component that also serves as a template in your static routes. Even if you don't actually use the component, only importing it triggers the message.

It doesn't seem to affect the final build but it's kind of annoying. If anybody could share their insight on this it'd be great!

Thanks!

**Comment by nddery** [comment]
Just realized that when those `FLUSH CHUNK` warnings appear, our built/exported site will flash white for a seconds upon loading (you see the static content, then when the JS app kicks in, the page render all white before re-rendering the content). This seems to appear right after the "Templates reloaded" console log appear. For us not changing `config.paths.dist` removes the warnings, as well as the flash upon loading.

**Comment by davidkleriga** [comment]
Has anyone found a solution to this? It seems to create strange problems in the build. For me, it's causing my meta tags to not display correctly. I am not importing a routable component in App, as described above. But I do dynamically import such a component in case the route doesn't exist yet (Dynamic Fallback)

**Comment by stefan-h-singer** [comment]
We're getting the same issue and we _must_ change the config.paths since we are building inside an AWS lambda (we only have write access to /tmp)

**Comment by stefan-h-singer** [comment]
I'd also like to add that we see the same issue that @nddery brought up above, the flash with the site rendered without content. Have a look at for instance https://www.lapointcamps.com. If I do a build locally with paths unchanged, I don't get the warnings during the build phase and no white flash when site transforms to react app.

**Comment by nddery** [comment]
Done some more digging on this and here's what I've found.

TL;DR; removing `ignoreBabelRename: true` from the `react-universal-component` options in [`generateTemplates`](https://github.com/react-static/react-static/blob/master/packages/react-static/src/static/generateTemplates.js#L26) seems to remove those warnings, when you've declared a `config.paths.dist` that points outside the `react-static` directory. It however creates the same problem if you've not declared that config option.

Looking at the web pack stats in [`exportRoute`](https://github.com/react-static/react-static/blob/master/packages/react-static/src/static/exportRoute.js#L168), I've noticed that chunks have dashes in them, while the chunk name we're looking for have slashes:

Part of `clientStats`:
```
{ main: [Object],
  '__react_static_root__-src-pages-404': [Object],
  '__react_static_root__-src-pages-about': [Object],
  '__react_static_root__-src-pages-blog': [Object],
  '__react_static_root__-src-pages-index': [Object] }
```

Chunk names:

```
[ '__react_static_root__/src/pages/about' ]
```

Seems like we're seeing the same problem as described [here](https://github.com/faceyspacey/webpack-flush-chunks/issues/74#issuecomment-427131503) - to which the maintainer said to add `ignoreBabelRename: true`.

With the `ignoreBabgelRename` option removed, building without setting the `config.paths.dist` now leads to the same FLUSH CHUNKS warning to appear, trying to find a chunk, with dashes (so the inverse), where the client stats have slashes in the chunk name.

> [FLUSH CHUNKS]: Unable to find __react_static_root__-src-pages-404

I don't know what the `ignoreBabelRename: true` does, I guess that's part of the next thing to figure out. We should probably also ensure we always use slashes or dashes. Chunk names are modified [here](https://github.com/react-static/react-static/blob/master/packages/react-static/src/static/exportRoute.js#L150-L155), maybe this has something to do with this.

**Comment by nddery** [comment]
Seems like commenting out https://github.com/react-static/react-static/blob/master/packages/react-static/src/static/generateTemplates.js#L36-L38 fixes the warnings in all cases.

**Comment by SleeplessByte** [comment]
I don't remember _why_ we needed that, but I believe it was because of duplicate names / paths whereas the files _were_ unique. 

> `ignoreBabelRename` is by default set to false which allows the plugin to attempt and name the dynamically imported chunk (replacing / with -). In more advanced scenarios where more granular control is required over the webpack chunk name, you should set this to true in addition to providing a function to chunkName to control chunk naming.

Related:

https://github.com/faceyspacey/babel-plugin-universal-import/pull/75
https://github.com/faceyspacey/react-universal-component/issues/137

**Comment by markatli** [comment]
Just checking if anyone has a solution for this issue?

We're seeing the same issue ("you see the static content, then when the JS app kicks in, the page render all white before re-rendering the content") as described by @nddery 

Not using config.paths.dist is not an option for us I'm afraid.

We don't care about the console.log "Templates reloaded" appearing, but the white flash and re-rendering of content is an issue for us.

Thanks!
